include/tuvx/linalg/linalg.inl

```diff
@@ -0,0 +1,83 @@
+
+#include <vector>
+namespace tuvx {
+namespace linalg {
+
+template <typename T>
+inline std::vector<T> dot(trid_mat<T> A, std::vector<T> x) {
+  int size = x.size();
+  std::vector<T> v(size);
+  v[0] = A.mdiag[0] * x[0] + A.udiag[0] * x[1];
+
+  int i = 0;
+  for (i = 1; i < x.size() - 1; i++) {
+    v[i] =
+        A.mdiag[i] * x[i] + A.udiag[i] * x[i + 1] + A.ldiag[i - 1] * x[i - 1];
+  }
+  v[i] = A.mdiag[i] * x[i] + A.ldiag[i - 1] * x[i - 1];
```

Comment:
```suggestion
  std::size_t i = 0;
  for (i = 1; i < size - 1; i++) {
    v[i] =
        A.mdiag[i] * x[i] + A.udiag[i] * x[i + 1] + A.ldiag[i - 1] * x[i - 1];
  }
  v[i] = A.mdiag[i] * x[i] + A.ldiag[i - 1] * x[i - 1];
```

Since you defined `size` above, may as well use it here